pretenders/common/http.py

```diff
@@ -213,10 +213,12 @@ def headers_match(self, headers):
             we're matching against.
         :return: True if the request is a match for headers and False if not.
         """
-        if self.headers:
-            for k, v in self.headers.items():
+        rq_headers = {k.lower(): v for k, v in headers.items()}
+        expected_headers = {k.lower(): v for k, v in self.headers.items()}
```

Comment:
This is fine but not too DRY. You could extract into a `normalize` function/lambda, e.g. something like:

```python
normalize = lambda headers: {k.lower(): v for k, v in headers.items()}
rq_headers = normalize(headers)
expected_headers = normalize(self.headers)
```